channeldb/channel.go

```diff
@@ -738,6 +758,26 @@ type OpenChannel struct {
 	// have private key isolation from lnd.
 	RevocationKeyLocator keychain.KeyLocator
 
+	// ZeroConf is a boolean that denotes whether the channel is a
+	// zero-conf channel.
+	ZeroConf bool
+
+	// OptionScidAlias is a boolean that denotes whether this channel has
+	// negotiated the option_scid_alias channel type and is not a zero-conf
```

Comment:
The channel type is different than the feature bit and is mutually exclusive with zero-conf

Reply:
is it? i didnt glean that from the spec. Seems reasonable to want a zero-conf channel that is also option_scid_alias

Reply:
Yes but i think both can still be set simultaneously

Reply:
Changed, only touched the funding part since I implemented a stricter version of the spec w/ some of the forwarding stuff. basically zero-conf and scid-alias channel types were the same except one could be public. still need to write an itest